### PR TITLE
Stm32mp1 bsec

### PR DESCRIPTION
**Pull request by patrickdelaunay**
Some updates in the BSEC driver used to OTPs access on STM32MP13x and STM32MP15x SoC family,
based on STMicroelectronics patches present in https://github.com/STMicroelectronics/optee_os.

The OTP descriptions are based on NVMEM cell binding.

The addition of the API "get otp by phandle()" is a preliminary steps for next series.

### CODE REVIEW

File: core/drivers/stm32_bsec.c [hunk 1]
```diff
@@ -625,22 +673,37 @@ TEE_Result stm32_bsec_find_otp_in_nvmem_layout(const char *name,
...
+	for (i = 0; i < nvmem_layout_count; i++) {
+		if (nvmem_layout[i].phandle != phandle)
+			continue;
+
+		return stm32_bsec_otp_setting(i, otp_id, otp_bit_offset,
+					      otp_bit_len);
 	}
 
-	DMSG("nvmem %s failed", name);
+	DMSG("nvmem %u not found", phandle);
```
Comment (tbourgoi) [inline core/drivers/stm32_bsec.c:703]: Maybe an EMSG could be use here instead.

Comment (patrickdelaunay) [reply core/drivers/stm32_bsec.c:703]: I think and "error" trace EMSG shoud be done by caller if it is needed.... it is note needed in BSEC driver himself

**Comment by tbourgoi** [comment]
Looks good to me, 
`Acked-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>`

File: core/include/drivers/stm32_bsec.h [hunk 2]
```diff
@@ -170,11 +170,13 @@ bool stm32_bsec_nsec_can_access_otp(uint32_t otp_id);
  * Find and get OTP location from its name.
  * @name: sub-node name to look up.
  * @otp_id: pointer to output OTP number or NULL.
+ * @otp_bit_offset: pointer to output OTP offset in bits or NULL.
```
Comment (etienne-lms) [inline core/include/drivers/stm32_bsec.h:173]: ` * @otp_bit_offset: pointer to output OTP bit offset in the NVMEM cell or NULL.`

File: core/drivers/stm32_bsec.c [hunk 3]
```diff
@@ -624,6 +638,7 @@ static size_t nvmem_layout_count;
 
 TEE_Result stm32_bsec_find_otp_in_nvmem_layout(const char *name,
 					       uint32_t *otp_id,
+					       uint8_t *otp_bit_offset,
```
Comment (etienne-lms) [inline core/drivers/stm32_bsec.c:664]: using a `uint8_t` pointer is not really needed, using native `size_t` is ok.
Not a strong opinion.

**Review by etienne-lms — COMMENTED** [review]
`Reviewed-by: Etienne Carriere <etienne.carriere@foss.st.com>` for commits
"drivers: stm32_bsec: fix stm32_bsec_find_otp_in_nvmem_layout()",
"drivers: stm32_bsec: add BSEC_DEN_ALL_MSK support",
"drivers: stm32_bsec: keep stm32_bsec_permanent_lock_otp() under flag" (please add your S-o-b tag to the commit message) and
"dts: stm32mp13: update part number NVMEM description" (please remove the Change-Id tag in this commit message).


Comments for commits "drivers: stm32_bsec: add support for bits property in the DT"
and "drivers: stm32_bsec: implement the get otp by phandle".

File: core/drivers/stm32_bsec.c (hunk 3 above)
Comment (patrickdelaunay) [reply core/drivers/stm32_bsec.c:664]: bit offset as limited range => 0...7, 
so uint8_t is enought
I plan to change type of bit_offset in nvmem_layout (I forget to do it)

File: core/drivers/stm32_bsec.c [hunk 4]
```diff
@@ -619,17 +619,40 @@ bool stm32_bsec_nsec_can_access_otp(uint32_t otp_id)
...
+	if (otp_id)
+		*otp_id = nvmem_layout[i].otp_id;
+
+	if (otp_bit_offset)
+		*otp_bit_offset = nvmem_layout[i].bit_offset;
+
+	if (otp_bit_len)
+		*otp_bit_len = nvmem_layout[i].bit_len;
+
+	DMSG("nvmem[%d] = %s at BSEC word %" PRIu32 " bits [%zu %zu]",
```
Comment (etienne-lms) [inline core/drivers/stm32_bsec.c:649]: s/%d/%zu/  for `size_t`
s/[%zu %zu]/[%"PRIu8" %zu]/  for `uint8_t`

File: core/drivers/stm32_bsec.c [hunk 5]
```diff
@@ -619,17 +619,40 @@ bool stm32_bsec_nsec_can_access_otp(uint32_t otp_id)
...
 	uint32_t otp_id;
 	uint8_t bit_offset;
 	size_t bit_len;
+	uint32_t phandle;
 };
 
 static struct nvmem_layout *nvmem_layout;
 static size_t nvmem_layout_count;
 
+static  TEE_Result stm32_bsec_otp_setting(size_t i,
```
Comment (etienne-lms) [inline core/drivers/stm32_bsec.c:635]: 2 space chars: remove one

**Review by etienne-lms — COMMENTED** [review]
minor comments

**Comment by patrickdelaunay** [comment]
> minor comments

corrected

**Comment by etienne-lms** [comment]
`Reviewed-by: Etienne Carriere <etienne.carriere@foss.st.com>` fior the while series.

**Comment by patrickdelaunay** [comment]
> `Reviewed-by: Etienne Carriere <etienne.carriere@foss.st.com>` fior the while series.

Review added

**Pull request merged**